pkg/api/norman/customization/aks/handler.go

```diff
@@ -154,6 +165,13 @@ func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credI
 	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
 	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
 
+	if cap.TenantID == "" {
+		cap.TenantID, err = getCachedTenantID(h.secretClient, cap.SubscriptionID, cc)
+		if err != nil {
+			logrus.Debugf("encountered error while updating secret, ignoring: %v", err)
```

Comment:
What if this error is around timestamp parsing or a timeout calling azure ? If we can't get the tenantID at all we should error out entirely, if we get the tenantID and error we can then just log.